17 update and logger from osl

nundinir opened the pull request:
Tested basic logging and sensor updating using OSL library. 

Added filing_utils.py file that can take in user arguments (follows the same experiment naming that we used to have, e.g. S101 pref dial 01312025 true). Also added a Logger object with certain dummy variables to track, as well as onboard sensor variables from the left exoboot.

VarunSatyadevShetty commented on src/utils/filing_utils.py:
```diff
@@ -0,0 +1,65 @@
+    
+    # set up argparser
+    parser = argparse.ArgumentParser(description="VAS experimental type (10X/trial type/trial cond/desc/usebackup):")
+    parser.add_argument("--sub", type=int, default=101, help="Subject ID (default: 101)", required=True)
```
I think for the argparser it would be be better to have the default subject name as "Test" or something which is not an actual subject, so we don't accidentally rewrite an actual data file, or make this a required argument.

VarunSatyadevShetty commented on main_controller.py:
```diff
@@ -0,0 +1,88 @@
+    LOG_LEVEL
+)
+
+def track_variables_for_logging(logger: Logger) -> None:
```
This function should look for the number of active sides and then track those variables (and not just be used for tracking the left exoboot values). Depending upon if only the left, right or both are active it should track those variables for logging.

VarunSatyadevShetty commented on main_controller.py:
```diff
@@ -0,0 +1,88 @@
+                
+                # record current values to buffer, log to file, then flush the buffer
+                logger.update()
+                logger.flush_buffer()
```
I would suggest not to do logger.flush_buffer() every single time we update the the logs. This will considerably slow down the process, if we try to write to the csv file so often. We can do the flush_buffer after a certain number (eg. 200) of timesteps, so that we are not writing to a csv file frequent but still updating the logs regularly incase of an issue with the exit log condition

VarunSatyadevShetty reviewed (changes requested):
Overall it looks good! 
Changes requested:
a) Changing the default subject number to be "Test" or making it a required argparser argument.
b) Calling logging flush_buffer() less frequently in the main loop.

nundinir commented:
Thanks for the feedback Varun!

I made the following changes: 
1. Made tracking variables flexible to actuator side. Now it will loop through any active actuators and name variables accordingly. I moved the method to the DephyExoboots class.
2. Removed flush_buffers method
3. Added argparser argument for tester subject

VarunSatyadevShetty reviewed (approved):
Great!! Thank you very much!
